searchlib/src/tests/queryeval/filter_wrapper/filter_wrapper_test.cpp

```diff
@@ -0,0 +1,124 @@
+            setDocId(20);
+        } else {
+            setAtEnd();
+        }
+    }
+
+    void doUnpack(uint32_t docid) override {
+        ++unpackCnt;
+        if (match != 0) {
+            _unpackedDocId = docid;
+        }
+    }
+};
+uint32_t DummyItr::seekCnt   = 0;
+uint32_t DummyItr::unpackCnt = 0;
+uint32_t DummyItr::dtorCnt   = 0;
+uint32_t DummyItr::_unpackedDocId = 0;
+
+
+TEST("filter wrapper forwards as expected") {
+    EXPECT_EQUAL(DummyItr::seekCnt, 0u);
+    EXPECT_EQUAL(DummyItr::unpackCnt, 0u);
+    EXPECT_EQUAL(DummyItr::dtorCnt, 0u);
+    { // without wrapper
+        TermFieldMatchData match;
+        DummyItr::_unpackedDocId = 0;
+        SearchIterator::UP search(new DummyItr(&match));
+        search->initFullRange();
+        EXPECT_EQUAL(DummyItr::_unpackedDocId, 0u);
+        EXPECT_TRUE(!search->seek(1u));
+        EXPECT_EQUAL(search->getDocId(), 10u);
+        EXPECT_TRUE(search->seek(10));
+        search->unpack(10);
+        EXPECT_EQUAL(DummyItr::_unpackedDocId, 10u);
+        EXPECT_TRUE(!search->seek(15));
+        EXPECT_EQUAL(search->getDocId(), 20u);
+        EXPECT_TRUE(search->seek(20));
+        search->unpack(20);
+        EXPECT_EQUAL(DummyItr::_unpackedDocId, 20u);
+        EXPECT_TRUE(!search->seek(25));
+        EXPECT_TRUE(search->isAtEnd());
+    }
+    EXPECT_EQUAL(DummyItr::seekCnt, 3u);
+    EXPECT_EQUAL(DummyItr::unpackCnt, 2u);
+    EXPECT_EQUAL(DummyItr::dtorCnt, 1u);
+    { // with wrapper
+        TermFieldMatchData match;
+        TermFieldMatchDataArray tfmda;
+        tfmda.add(&match);
+        DummyItr::_unpackedDocId = 0;
+        auto search = std::make_unique<FilterWrapper>(1);
+        auto to_wrap = std::make_unique<DummyItr>(search->tfmda()[0]);
+        search->wrap(std::move(to_wrap));
+        search->initFullRange();
+        EXPECT_EQUAL(DummyItr::_unpackedDocId, 0u);
+        EXPECT_TRUE(!search->seek(1u));
+        EXPECT_EQUAL(search->getDocId(), 10u);
+        EXPECT_TRUE(search->seek(10));
+        search->unpack(10);
+        EXPECT_EQUAL(DummyItr::_unpackedDocId, 0u);
+        EXPECT_TRUE(!search->seek(15));
+        EXPECT_EQUAL(search->getDocId(), 20u);
+        EXPECT_TRUE(search->seek(20));
+        search->unpack(20);
+        EXPECT_EQUAL(DummyItr::_unpackedDocId, 0u);
+        EXPECT_TRUE(!search->seek(25));
+        EXPECT_TRUE(search->isAtEnd());
+    }
+    EXPECT_EQUAL(DummyItr::seekCnt, 6u);
+    EXPECT_EQUAL(DummyItr::unpackCnt, 2u);
+    EXPECT_EQUAL(DummyItr::dtorCnt, 2u);
+}
+
+class Verifier : public search::test::SearchIteratorVerifier {
+public:
+    ~Verifier();
+    SearchIterator::UP create(bool strict) const override {
+        auto search = std::make_unique<FilterWrapper>(1);
+        search->wrap(createIterator(getExpectedDocIds(), strict));
+        return search;
+    }
+};
+
+Verifier::~Verifier() {}
+
+TEST("Test that filter wrapper iterators adheres to SearchIterator requirements") {
+    Verifier searchIteratorVerifier;
```

Comment:
I guess this is the reason for using testkit instead of gtest?

Reply:
reason for all your comments: this is an almost-exact copy of booleanmatchiteratorwrapper unit test. i'll look at modernizing both of these tests (and maybe put them in the same file, or somehow avoid so much code duplication) soon.